docs/source/plugin/bot/channels.rst

```diff
@@ -0,0 +1,146 @@
+can iterate over to get all the users::
+
+    for nick, user in channel.users.items():
+        # do something with the nick and the user
+
+You can access one user in a channel with its nick::
+
+    user = channel.users['Nickname']
+
+With the ``trigger`` object, you can also access the user object directly::
+
+    user = channel.users[trigger.nick]
+
+Manage channels
+===============
+
+Like any IRC client, Sopel uses IRC messages to join or leave channels, to kick
+users or to set channel modes. It provides methods you can use to send those
+commands:
+
+* :meth:`bot.join() <sopel.irc.AbstractBot.join>`: to join a channel
+* :meth:`bot.part() <sopel.irc.AbstractBot.part>`: to leave a channel
+* :meth:`bot.kick() <sopel.irc.AbstractBot.kick>`: to kick a user from a
+  channel (this may requires some extra privileges)
+* :meth:`bot.write() <sopel.irc.AbstractBot.write>`: to send any message to IRC
+
+Join & Part
+-----------
+
+If you want the bot to join or leave a channel, the solution is
+straightforward::
+
+    bot.join('#channel_name')
+    bot.part('#channel_name')
+
+For example, you can recreate the ``.join #channel`` command like so::
+
+    @plugin.commands('join')
+    @plugin.require_admin()
+    def my_custom_join(bot, trigger):
+        channel, key = trigger.group(3), trigger.group(4)
+        bot.join(channel, password=key)
+
+In this example, the bot sends the following IRC message to the server:
+
+.. code-block:: irc
+
+    JOIN #channel
+
+When the server replies to that message, Sopel will automatically update its
+list of known channels and may send other request to know more about its users.
+
+However, it won't add the channel to its configuration: you will have to do
+that yourself by adding it to
+:attr:`bot.settings.core.channels <sopel.config.core_section.CoreSection.channels>`
+if you choose to do so.
+
+The ``part`` method works exactly like the ``join`` method: it sends a message
+to the IRC server and it is up to you to decide if you want to save or not
+that information in the settings—in that case by removing it from the list.
+
+.. warning::
+
+    You should always consider security when allowing the bot to join or leave
+    a channel: in this example, only admins are allowed to use the ``.join``
+    command thanks to the :func:`~sopel.plugin.require_admin` decorator.
+
+Kick users
+----------
+
+Sometimes you don't want a user in your channel. The IRC message would be:
+
+.. code-block:: irc
+
+    KICK #channel xnaas :You know why!
+
+You can use :meth:`bot.kick() <sopel.irc.AbstractBot.kick>` to achieve the same
+result::
+
+    bot.kick('xnaas', '#channel', text='You know why!')
+
+.. warning::
+
+    Be responsible: you should ensure that your plugin limits abuse as much as
+    possible, either by limiting who can trigger a kick (by checking
+    :doc:`privileges </plugin/privileges>`), or by limiting kicks to an
+    explicit set of reasons and/or actions.
```

Comment:
```suggestion
    Be responsible: you should ensure that your plugin limits the potential for abuse as much as
    possible. For example, consider limiting who can trigger a kick by checking
    :doc:`privileges </plugin/privileges>`, and/or limiting kicks to an
    explicit set of reasons.
```

Lazy non-rewrap since this is a pretty big change that you'll probably want to tweak anyway before accepting, if at all.